### PR TITLE
calypso-config: Extract `calypso/config` into new @automattic/calypso-config package

### PR DESCRIPTION
**Pull request by sarayourfriend**
#### Changes proposed in this Pull Request

* Extracts `calypso/config` into its own `@automattic/calypso-config` package.

#### Testing instructions

* Ensure build is successful
* Ensure unit tests pass
* Ensure you can still enable flags via cookies and the `flags` parameter
* Smoke test the app to make sure configuration is still working

Note: There are various linting errors that I'm ignoring as out of scope for this PR. None of them are new.

### CODE REVIEW

**Comment by sirbrillig** [comment]
> packages/calypso-config/src/index.ts:11:33 - error TS2307: Cannot find module '@automattic/create-calypso-config' or its corresponding type declarations.
> 11 import type { ConfigData } from '@automattic/create-calypso-config';

Looks like there's some errors where the wrong module name is imported, but overall this is very exciting. `config` has been one of the main blockers to moving many checkout-related things to their own packages.

**Review by sirbrillig — APPROVED** [review]
I'll approve on behalf of checkout and payments management, though I can't speak to other areas. I also kicked off the full e2e test suite because this touches so many parts of the codebase I imagine it would be helpful to see if those pass.

**Comment by sarayourfriend** [comment]
Good call, thanks @sirbrillig

File: client/server/boot/index.js [hunk 1]
```diff
@@ -11,6 +11,7 @@ import userAgent from 'express-useragent';
  * Internal dependencies
  */
 import analytics from 'calypso/server/lib/analytics';
+console.log( '==================HERE==================' );
```
Comment (lynnbgriffith) [inline client/server/boot/index.js]: Found this hiding here.

**Review by lynnbgriffith — APPROVED** [review]
Overall, this looks good. Was the original motivation the use of TypeScript? In that regard, the config API works well with your changes. I noted a stray `console.log` in the review. Assuming e2e tests look good and that's tidied up, I'm happy with this.

**Comment by sarayourfriend** [comment]
Whoops, thanks for catching that @griffbrad. And yup, wanted to transform this TypeScript and also make it available to other packages without having to depend on `client` (looks like the `@automattic/components` package already did do that!)

**Pull request merged**